python/mscclpp_benchmark/allreduce.cu

```diff
@@ -788,6 +788,24 @@ extern "C" __global__ void __launch_bounds__(1024, 1)
 
 #if defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 900
 
+// Barrier among all devices followed by a memory fence
+// Should be called by all threads on all devices
+// Assumes \p num_threads_per_block >= \p num_ranks
+__device__ void barrier(mscclpp::SmDevice2DeviceSemaphoreDeviceHandle* semaphores, int thread_id, int block_id,
+                        int num_threads_per_block, int num_blocks, int num_ranks) {
```

Comment:
Seems `num_threads_per_block` is not used

Reply:
It is not